UOP1_Project/Assets/Scripts/Editor/GameSceneEditor.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+[CustomEditor(typeof(GameScene), true)]
+public class GameSceneEditor : Editor
+{
+    #region UI_Warnings
+
+    private string noScenesWarning = "There are no scenes set for this level yet! Add a new scene with the dropdown above";
+    
+    #endregion
+    
+    // it holds a list of properties to hide from basic inspector
+    private static readonly string[] ExcludedProperties = {"m_Script", "sceneName"};
+    
+    private string[] sceneList;
+    private GameScene gameSceneTarget;
+
+    private void OnEnable()
+    {
+        gameSceneTarget = target as GameScene;
+    }
+
+    public override void OnInspectorGUI()
+    {
+        PopulateScenePicker();
```

Comment:
Why this function is inside this loop?
Can it be on `OnEnable` function?

Reply:
Done